tests/integration_tests/test_catalyst.py

```diff
@@ -0,0 +1,53 @@
+import sys
+
+import pytest
+import torch
+
+import optuna
+from optuna.integration import CatalystPruningCallback
+from optuna.testing.integration import DeterministicPruner
+
+catalyst = pytest.importorskip("catalyst")
+
+
+@pytest.mark.skipif(sys.version_info < (3, 6), reason="catalyst requires python3.6 or higher")
+def test_catalyst_pruning_callback():
+    # type: () -> None
+
+    data = torch.zeros(3, 4, dtype=torch.float32)
+    target = torch.zeros(3, dtype=torch.float32)
+    dataset = torch.utils.data.TensorDataset(data, target)
+
+    loaders = {
+        "train": torch.utils.data.DataLoader(dataset, batch_size=1),
+        "valid": torch.utils.data.DataLoader(dataset, batch_size=1),
+    }
+
+    def objective(trial):
+        # type: (optuna.trial.Trial) -> float
+        model = torch.nn.Linear(4, 1)
+        criterion = torch.nn.MSELoss()
+        optimizer = torch.optim.Adam(model.parameters())
+
+        runner = catalyst.dl.SupervisedRunner()
+        runner.train(
+            model=model,
+            criterion=criterion,
+            optimizer=optimizer,
+            loaders=loaders,
+            logdir="./logdir",
```

Comment:
Wondering if this test isn't creating files. If so, we might want to make sure it's cleaned up or use some temporary directory.